tendrl/node_agent/manager/manager.py

```diff
@@ -214,6 +215,29 @@ def on_pull(self, raw_data):
                     node_id=raw_data["node_id"],
                 )
             )
+        if "disks" in raw_data:
+            LOG.info("on_pull, Updating disks")
+            try:
+                self.etcd_client.delete(
+                    ("nodes/%s/Disks") % raw_data["node_id"], recursive=True)
+            except etcd.EtcdKeyNotFound as ex:
+                LOG.debug("Given key is not present in etcd . %s", ex)
+            disks = raw_data['disks']
+            if "disks" in disks:
+                for disk in disks['disks']:
+                    disk['node_id'] = raw_data['node_id']
+                    disk_obj = Disk(disk)
+                    disk_json = disk_obj.to_json_string()
+                    self.etcd_client.write((disk_obj.__name__) % (
+                        raw_data["node_id"], disk['disk_id']), disk_json)
+            if "used_disks_id" in disks:
+                for disk in disks['used_disks_id']:
+                    self.etcd_client.write(("nodes/%s/Disks/used/%s") % (
+                        raw_data["node_id"], disk), "")
```

Comment:
@GowthamShanmugam I have repeatedly requested, implement classes for objects defined under Tendrl (class FreeDisk, class UsedDisk) and not simply write these objects to etcd, anyways Ill take this up in another issue, this is too time consuming